org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java

```diff
@@ -354,6 +357,12 @@ public static <T> boolean isEmpty(List<T> list) {
 	private static Collection<File> classpathEntriesToFiles(JavaProject project, Predicate<IClasspathEntry> select) {
 		try {
 			LinkedHashSet<File> res = new LinkedHashSet<>();
+			if (project.getModuleDescription() == null) {
+				IPath outputLocation = project.getOutputLocation();
+				if (outputLocation != null) {
+					addPath(project, outputLocation, res);
+				}
+			}
```

Comment:
Adding the project output to classpath should already be done with
```
else if (select.test(current)) {
	IPath path = current.getPath();
	addPath(project, path, res);
}
```
around line 388.
So I don't think it's necessary as part of this change.

Reply:
> This code adds the source entry but not the output location.

Oops, I thought I had fixed that; thanks for spotting it.
Can you please submit it as a separate PR as it's desired independently for lombok?

Reply:
I have created https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/pull/1006 and updated this PR.

Reply:
#1006 has been merged. I have rebased this PR to include #1006.